docs/release.md

```diff
@@ -0,0 +1,78 @@
+# Release
+
+> [!NOTE]
+> This document is based on the initial learnings from the release process and is therefore subject to change.
+
+## Preparing the release
+
+For each release, a milestone needs to be in place. All PRs that are part of the release need to be assigned to 
+the milestone.
+
+This will also make it easier to filter for the correct commits, that should be cherry-picked later.
+
+### Create an issue
+
+To track the progress of the release, we need to create an issue. Make use of the `Release` issue template.
+
+### Patch release
+
+A patch release can only include changes that are backwards compatible with the previous release. 
+This includes bug fixes, documentation updates, and other minor changes.
+
+New features and breaking changes must be part of a minor release.
+
+### Minor release
+
+A minor release includes all changes from the last minor version to the next one. 
+This also includes new features and breaking changes.
+
+The following Tasks need to be done before creating a new release:
+
+- Update metadata & clusterctl-settings.
+- Update version in e2e config.
+- Update docs (compatibility table; usage etc).
+
+### Milestones
+
+With `Milestones` we want to make it easier to track the progress of a release.
+
+### Cherry-Picking
+
+> [!NOTE]
+> As we do not yet have `Prow`, we need to do the cherry-picking manually.
```

Comment:
I think, it's better to not mention Prow here. 

However, i still think we need to label PRs that will go to next patch release. 
As in CAPI. 
E.g. release-0.2

Reply:
Just use milestones

Reply:
As you wish